keydata_legacy_test.go

```diff
@@ -0,0 +1,334 @@
+package secboot
```

Comment:
This should be in package `secboot_test`. I think moving this will create a naming conflict though because there are 2 `keyDataTestBase` types.

Reply:
I refactored the legacy keyData test suite.